frontend/src/main/scala/UnofficialDeferment.scala

```diff
@@ -153,19 +78,84 @@ class UnofficialDeferment() extends js.Object {
   }
 
   def getBandsForTaxYear(taxYear: Int): js.Array[scala.scalajs.js.Object] = {
-    val taxYearBands =  config.get(taxYear).getOrElse(
+    val taxYearBands =  config.unofficialDeferment.get(taxYear).getOrElse(
       sys.error(s"Could not find config for tax year $taxYear")
     )
 
     taxYearBands.bandLimits.toJSArray.map{ band =>
       new js.Object {
         val label = band.toLabel
-        val amount = band.value
+        val amount = band.value.doubleValue()
       }
     }
   }
 }
 
+object UnofficialDeferment {
+
+  implicit class Class1BandOps(private val b: Class1Band) extends AnyVal {
+    def toLabel: String = b match {
+      case BelowLEL => "LEL"
+      case LELToET =>  "LEL - ET"
+      case LELToPT =>  "LEL - PT"
+      case PTToUAP =>  "PT - UAP"
+      case PTToUEL =>  "PT - UEL"
+      case ETToUEL =>  "ET - UEL"
+      case UAPToUEL => "UAP - UEL"
+      case AboveUEL => "UEL"
+    }
+
+  }
+
+  def labelToClass1Band(label: String): Option[Class1Band] = label match {
+    case "LEL"                => Some(BelowLEL)
+    case "LEL - ET"            => Some(LELToET)
+    case "LEL - PT"            => Some(LELToPT)
+    case "PT - UAP"            => Some(PTToUAP)
+    case "PT - UEL"            => Some(PTToUEL)
+    case "ET - UEL"            => Some(ETToUEL)
+    case "UAP - UEL"           => Some(UAPToUEL)
+    case  "UEL"                => Some(AboveUEL)
+    case _ => None
+
+
+  }
+
+  implicit class Class1BandLimitOps(private val l: Class1BandLimit) extends AnyVal {
+    def toLabel: String = l match {
+      case _: LEL  => "Lower earning limit"
+      case _: PT =>   "Primary threshold"
+      case _: ET =>   "Earning threshold"
+      case _: UAP  => "Upper accrual point"
+      case _: UEL  => "Upper earning limit"
+    }
+
+  }
+
+  def labelToClass1BandLimit(label: String, amount: Double): Option[Class1BandLimit] = label match {
+    case   "Lower earning limit"    => Some(LEL(amount))
+    case   "Primary threshold"      => Some(PT(amount))
+    case   "Earning threshold"      => Some(ET(amount))
+    case   "Upper accrual point"    => Some(UAP(amount))
+    case   "Upper earning limit"    => Some(UEL(amount))
+    case _ => None
+  }
+
+
+
+  def class1bandLimitFromLabel(label: String, amount: Double) = label match {
+    case "Lower earning limit"         => LEL(amount)
+    case "Primary threshold"           => PT(amount)
+    case "Earning threshold"           => ET(amount)
+    case "Upper accrual point"         => UAP(amount)
+    case "Upper earning limit"         => UEL(amount)
+    case other                         => sys.error(s"Could not understand label $other")
```

Comment:
Is this not `labelToClass1BandLimit(label, amount).getOrElse(sys.error(...))`?

Reply:
whoops, I meant to take that out. It's gone now